### DISCUSSION TITLE
How do you add a checkpoint from civitai? Colab

### DISCUSSION THREAD

**Discussion by floxdaw** [lllyasviel/Fooocus#1228] [Q&A] [answered]
I am trying to add the checkpoint from absolute reality (https://civitai.com/models/81458/absolutereality) to the Colab UI, but I cant seem to find the right folder. The .safetensor file keeps getting corrupted, and when I add the line:

`!wget -c https://civitai.com/api/download/models/132760 -O ./models/loras/absolutereality.safetensor `

Nothing changes on the advanced ->model page, even after presing the "Refresh all files" button.

**Comment by tbuyle** [comment]
Checkpoints go to ./models/checkpoints

**Reply by AndreiLungu64** [reply]
After uploading, how do I use it? I select it in the base models? (i'm doing this and it is returning this error: NotImplementedError: You have selected base model other than SDXL. This is not supported yet.)

**Reply by Sven-Lyco** [reply]
You must make sure that you use SDXL. It is best to filter for SDXL and checkpoints in CivitAI. Then you can select it

**Reply by tbuyle** [reply]
As pointed by @Sven-Lyco, you need to use SDXL-based model.

absolute reality is a SD 1.5 based checkpoints and won't work with Fooocus

![image](https://github.com/lllyasviel/Fooocus/assets/467350/dd234a72-fbd2-4d47-96dc-9ba19f6d782e)

**Reply by AndreiLungu64** [reply]
Thanks!

**Reply by AndreiLungu64** [reply]
![image](https://github.com/lllyasviel/Fooocus/assets/118302943/35f730ce-00d3-4716-b3cf-4f3979839f69)
Every version of SDXL will work?

**Reply by Sven-Lyco** [reply]
should work

**Reply by AndreiLungu64** [reply]
After downloading https://civitai.com/models/139562/realvisxl-v20 and selecting it in the base model I get this error when trying to generate. Whay is the file corrupted? (the command I used was: !wget -c https://civitai.com/api/download/models/169921 --content-disposition -O /content/Fooocus/models/checkpoints/realvisxlV20_v20Bakedvae.safetensors)

Traceback (most recent call last):
File "/content/Fooocus/modules/patch.py", line 512, in loader
result = original_loader(*args, **kwargs)
File "/usr/local/lib/python3.10/dist-packages/safetensors/torch.py", line 259, in load_file
with safe_open(filename, framework="pt", device=device) as f:
safetensors_rust.SafetensorError: Error while deserializing header: HeaderTooLarge

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "/content/Fooocus/modules/async_worker.py", line 803, in worker
handler(task)
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/content/Fooocus/modules/async_worker.py", line 354, in handler
pipeline.refresh_everything(refiner_model_name=refiner_model_name, base_model_name=base_model_name,
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/content/Fooocus/modules/default_pipeline.py", line 233, in refresh_everything
refresh_base_model(base_model_name)
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/content/Fooocus/modules/default_pipeline.py", line 69, in refresh_base_model
model_base = core.load_model(filename)
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/content/Fooocus/modules/core.py", line 152, in load_model
unet, clip, vae, clip_vision = load_checkpoint_guess_config(ckpt_filename, embedding_directory=path_embeddings)
File "/content/Fooocus/backend/headless/fcbh/sd.py", line 420, in load_checkpoint_guess_config
sd = fcbh.utils.load_torch_file(ckpt_path)
File "/content/Fooocus/backend/headless/fcbh/utils.py", line 13, in load_torch_file
sd = safetensors.torch.load_file(ckpt, device=device.type)
File "/content/Fooocus/modules/patch.py", line 528, in loader
raise ValueError(exp)
ValueError: Error while deserializing header: HeaderTooLarge
File corrupted: /content/Fooocus/models/checkpoints/realvisxlV20_v20Bakedvae.safetensors
Fooocus has tried to move the corrupted file to /content/Fooocus/models/checkpoints/realvisxlV20_v20Bakedvae.safetensors.corrupted
You may try again now and Fooocus will download models again.

Exception in thread Thread-3 (worker):
Traceback (most recent call last):
File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
self.run()
File "/usr/lib/python3.10/threading.py", line 953, in run
self._target(*self._args, **self._kwargs)
File "/content/Fooocus/modules/async_worker.py", line 809, in worker
pipeline.prepare_text_encoder(async_call=True)
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/content/Fooocus/modules/default_pipeline.py", line 210, in prepare_text_encoder
assert_model_integrity()
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "/content/Fooocus/modules/default_pipeline.py", line 49, in assert_model_integrity
if not isinstance(model_base.unet_with_lora.model, SDXL):
AttributeError: 'NoneType' object has no attribute 'model'
Total time: 0.55 seconds

**Reply by floxdaw** [reply]
I am also getting 
` AttributeError: 'NoneType' object has no attribute 'model' `
error, on a few different checkpoints. 

Does anyone know why this is?

**Reply by floxdaw** [reply, marked as answer]
I think I figured it out. You need to copy the download link directly and not just guess the model ID.

Copy and paste the download link:
![Screenshot 2023-12-07 222006](https://github.com/lllyasviel/Fooocus/assets/63487229/8572b57b-229a-4bd9-b155-e35ab0f4c367)

It turns out that the website URL and API can have different addresses, so make sure to copy the model ID straight from the download link, not the web page URL.

Once you do that, using a command similar to:

` !wget -c https://civitai.com/api/download/models/231175 -O ./models/checkpoints/photopedia.safetensors`

Using the link you copied before as the first link.
Make sure you past the wget lines of code immediately above the last line of code in the Fooocus template

If its not working, go to Runtime -> disconnect and delete runtime
This let you start a new session without any corrupt files. Then repeat the above step.

Using this method I got it to work

**Reply by edwardk1234** [reply]
Tried what Floxdaw said above. Works!

**Reply by maizhouzi** [reply]
`wget` only supports single-threaded downloads, which are relatively slow. I use `aria2` as a replacement, which significantly improves download speed. For example:
```
!apt-get install aria2
%shell aria2c -x 16 -s 16 -o "/models/checkpoints/NAME_OF_MODEL.safetensors" "https://civitai.com/api/download/models/NUMBER_OF_MODEL?type=Model&format=SafeTensor&size=full&fp=fp16"
```
replace the `NAME_OF_MODEL` and `NUMBER_OF_MODEL`

**Comment by dpv20** [comment]
how do u guys do it on windows? i cant use wget and i thought it was as simple as downloading it from the page and past it in ./models/checkpoints

**Reply by Sven-Lyco** [reply]
Yes, you can do it like that

**Reply by mashb1t** [reply]
the wget approach is good for Colab because this can be automated in the Jupyter Notebook. You do not have to do this locally!

**Reply by dpv20** [reply]
thats for linux, im on windows

**Reply by mashb1t** [reply]
not necessarily, fyi you can also use wget on Windows: https://gnuwin32.sourceforge.net/packages/wget.htm

**Reply by floxdaw** [reply]
The answer thread was also done on windows 11

**Comment by rayronvictor** [comment]
Change your `config.txt` to include checkpoints and restart on Colab:

For example, my config.txt looks like this:

```js
{
    // ... using juggernautXL v8 and v6
    "checkpoint_downloads": {
        "juggernautXL_version6Rundiffusion.safetensors": "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/juggernautXL_version6Rundiffusion.safetensors",
        "juggernautXL_v8Rundiffusion.safetensors": "https://civitai.com/api/download/models/288982"
    }
}
```

**Reply by Kyl329** [reply]
isnt it easier to mount the drive and shutil the model from the drive?

from my experience, download speed for other models other than juggernaut V6 tank

**Reply by mashb1t** [reply]
Not everybody is using google drive.
Not everybody has space left there or the bandwidth to first download the model locally and then upload again on drive.
=> "easier" is subjective in this case.

The speed completely depends on the CDN or the hoster where to download from and Colab network bandwidth restrictions (i know of none).

**Comment by eddyizm** [comment]
I found it easier to set up a preset with the model I want from there then on the free Google colab, I just start it with my new preset. 

This is easier if you create your own fork.

**Comment by Kyl329** [comment]
Care to share the direct model file code for the newbies, like me?

**Reply by eddyizm** [reply]
@Kyl329 Sure. 
Here's a direct [link ](https://github.com/eddyizm/Fooocus/blob/change_default_model/presets/realistic_fantasy.json) to the json preset I created this morning and tested in colab. 
Make sure that is in your preset folder then call it 
`!python entry_with_update.py --preset realistic_fantasy --share --always-high-vram --all-in-fp16` where the preset is the critical part.

EDIT: forgot to add the preset pulls in this model - https://civitai.com/models/136220/realistic-fantasy-mix-sdxl - but making your own to call any SDXL model from civitia would be trivial once you get this one working. 

Let me know if you have any questions.

**Comment by KyleeC137** [comment]
> Change your `config.txt` to include checkpoints and restart on Colab:
> 
> For example, my config.txt looks like this:
> 
> ```js
> {
>     // ... using juggernautXL v8 and v6
>     "checkpoint_downloads": {
>         "juggernautXL_version6Rundiffusion.safetensors": "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/juggernautXL_version6Rundiffusion.safetensors",
>         "juggernautXL_v8Rundiffusion.safetensors": "https://civitai.com/api/download/models/288982"
>     }
> }
> ```



> @Kyl329 Sure. Here's a direct [link ](https://github.com/eddyizm/Fooocus/blob/change_default_model/presets/realistic_fantasy.json) to the json preset I created this morning and tested in colab. Make sure that is in your preset folder then call it `!python entry_with_update.py --preset realistic_fantasy --share --always-high-vram --all-in-fp16` where the preset is the critical part.
> 
> EDIT: forgot to add the preset pulls in this model - https://civitai.com/models/136220/realistic-fantasy-mix-sdxl - but making your own to call any SDXL model from civitia would be trivial once you get this one working.
> 
> Let me know if you have any questions.


Juggernaut v8 download speeds are incredibly slow using this method, any ideas why?

**Reply by mashb1t** [reply]
> The model download URL might (and probably should due to higher speeds) also be changed to lllyasviel huggingface namespace.

See https://github.com/lllyasviel/Fooocus/pull/1785

Civitai.com is prone to limit speeds as user traffic must be enormous. On certain times of the day civitai can also be just plain offline / unavailable. HuggingFace is much more stable when you want to reliably download content + offers higher speeds due to (i assume) using a better / different CDN.

**Comment by shaitanzx** [comment]
you can use new fork - Fooocus_extend (https://github.com/shaitanzx/Fooocus_extend)
